cli/compose/convert/volume.go

```diff
@@ -88,3 +67,77 @@ func convertVolumeToMount(
 	// Named volumes
 	return result, nil
 }
+
+func handleBindToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type bind")
```

Comment:
On error-situations, we should not return a `result` with values; the result should be "empty" Mount, e.g.;

```go
return mount.Mount{}, errors.New("volume options are incompatible with type bind")
```

but also see my other comment on moving the `result` mount outside of these functions; it's likely better to construct the result at the end of each `handleXX` function